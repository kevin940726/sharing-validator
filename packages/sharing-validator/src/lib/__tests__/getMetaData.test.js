import getMetadata from "../getMetaData";

test("spotify", async () => {
  const meta = await getMetadata(
    "https://open.spotify.com/track/4u7EnebtmKWzUH433cf5Qv"
  );

  expect(meta).toMatchSnapshot();
});
