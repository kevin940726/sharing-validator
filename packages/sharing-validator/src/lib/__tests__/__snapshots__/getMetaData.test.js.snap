// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spotify 1`] = `
Array [
  Array [
    "og:title",
    "Bohemian Rhapsody - Remastered 2011",
  ],
  Array [
    "og:description",
    "Bohemian Rhapsody - Remastered 2011, a song by Queen on Spotify",
  ],
  Array [
    "og:url",
    "https://open.spotify.com/track/4u7EnebtmKWzUH433cf5Qv",
  ],
  Array [
    "og:image",
    "https://i.scdn.co/image/356e29a0f112a4835f7b22f5e586a1907eae2031",
  ],
  Array [
    "og:type",
    "music.song",
  ],
  Array [
    "music:duration",
    "354",
  ],
  Array [
    "music:album",
    "https://open.spotify.com/album/1GbtB4zTqAsyfZEsm1RZfx",
  ],
  Array [
    "music:album:track",
    "11",
  ],
  Array [
    "music:musician",
    "https://open.spotify.com/artist/1dfeR4HaWDbWqFHLkxsg1d",
  ],
  Array [
    "music:release_date",
    "1975-11-21",
  ],
  Array [
    "og:restrictions:country:allowed",
    "AD",
  ],
  Array [
    "og:restrictions:country:allowed",
    "AE",
  ],
  Array [
    "og:restrictions:country:allowed",
    "AR",
  ],
  Array [
    "og:restrictions:country:allowed",
    "AT",
  ],
  Array [
    "og:restrictions:country:allowed",
    "AU",
  ],
  Array [
    "og:restrictions:country:allowed",
    "BE",
  ],
  Array [
    "og:restrictions:country:allowed",
    "BG",
  ],
  Array [
    "og:restrictions:country:allowed",
    "BH",
  ],
  Array [
    "og:restrictions:country:allowed",
    "BO",
  ],
  Array [
    "og:restrictions:country:allowed",
    "BR",
  ],
  Array [
    "og:restrictions:country:allowed",
    "CA",
  ],
  Array [
    "og:restrictions:country:allowed",
    "CH",
  ],
  Array [
    "og:restrictions:country:allowed",
    "CL",
  ],
  Array [
    "og:restrictions:country:allowed",
    "CO",
  ],
  Array [
    "og:restrictions:country:allowed",
    "CR",
  ],
  Array [
    "og:restrictions:country:allowed",
    "CY",
  ],
  Array [
    "og:restrictions:country:allowed",
    "CZ",
  ],
  Array [
    "og:restrictions:country:allowed",
    "DE",
  ],
  Array [
    "og:restrictions:country:allowed",
    "DK",
  ],
  Array [
    "og:restrictions:country:allowed",
    "DO",
  ],
  Array [
    "og:restrictions:country:allowed",
    "DZ",
  ],
  Array [
    "og:restrictions:country:allowed",
    "EC",
  ],
  Array [
    "og:restrictions:country:allowed",
    "EE",
  ],
  Array [
    "og:restrictions:country:allowed",
    "EG",
  ],
  Array [
    "og:restrictions:country:allowed",
    "ES",
  ],
  Array [
    "og:restrictions:country:allowed",
    "FI",
  ],
  Array [
    "og:restrictions:country:allowed",
    "FR",
  ],
  Array [
    "og:restrictions:country:allowed",
    "GB",
  ],
  Array [
    "og:restrictions:country:allowed",
    "GR",
  ],
  Array [
    "og:restrictions:country:allowed",
    "GT",
  ],
  Array [
    "og:restrictions:country:allowed",
    "HK",
  ],
  Array [
    "og:restrictions:country:allowed",
    "HN",
  ],
  Array [
    "og:restrictions:country:allowed",
    "HU",
  ],
  Array [
    "og:restrictions:country:allowed",
    "ID",
  ],
  Array [
    "og:restrictions:country:allowed",
    "IE",
  ],
  Array [
    "og:restrictions:country:allowed",
    "IL",
  ],
  Array [
    "og:restrictions:country:allowed",
    "IN",
  ],
  Array [
    "og:restrictions:country:allowed",
    "IS",
  ],
  Array [
    "og:restrictions:country:allowed",
    "IT",
  ],
  Array [
    "og:restrictions:country:allowed",
    "JO",
  ],
  Array [
    "og:restrictions:country:allowed",
    "JP",
  ],
  Array [
    "og:restrictions:country:allowed",
    "KW",
  ],
  Array [
    "og:restrictions:country:allowed",
    "LB",
  ],
  Array [
    "og:restrictions:country:allowed",
    "LI",
  ],
  Array [
    "og:restrictions:country:allowed",
    "LT",
  ],
  Array [
    "og:restrictions:country:allowed",
    "LU",
  ],
  Array [
    "og:restrictions:country:allowed",
    "LV",
  ],
  Array [
    "og:restrictions:country:allowed",
    "MA",
  ],
  Array [
    "og:restrictions:country:allowed",
    "MC",
  ],
  Array [
    "og:restrictions:country:allowed",
    "MT",
  ],
  Array [
    "og:restrictions:country:allowed",
    "MX",
  ],
  Array [
    "og:restrictions:country:allowed",
    "MY",
  ],
  Array [
    "og:restrictions:country:allowed",
    "NI",
  ],
  Array [
    "og:restrictions:country:allowed",
    "NL",
  ],
  Array [
    "og:restrictions:country:allowed",
    "NO",
  ],
  Array [
    "og:restrictions:country:allowed",
    "NZ",
  ],
  Array [
    "og:restrictions:country:allowed",
    "OM",
  ],
  Array [
    "og:restrictions:country:allowed",
    "PA",
  ],
  Array [
    "og:restrictions:country:allowed",
    "PE",
  ],
  Array [
    "og:restrictions:country:allowed",
    "PH",
  ],
  Array [
    "og:restrictions:country:allowed",
    "PL",
  ],
  Array [
    "og:restrictions:country:allowed",
    "PS",
  ],
  Array [
    "og:restrictions:country:allowed",
    "PT",
  ],
  Array [
    "og:restrictions:country:allowed",
    "PY",
  ],
  Array [
    "og:restrictions:country:allowed",
    "QA",
  ],
  Array [
    "og:restrictions:country:allowed",
    "RO",
  ],
  Array [
    "og:restrictions:country:allowed",
    "SA",
  ],
  Array [
    "og:restrictions:country:allowed",
    "SE",
  ],
  Array [
    "og:restrictions:country:allowed",
    "SG",
  ],
  Array [
    "og:restrictions:country:allowed",
    "SK",
  ],
  Array [
    "og:restrictions:country:allowed",
    "SV",
  ],
  Array [
    "og:restrictions:country:allowed",
    "TH",
  ],
  Array [
    "og:restrictions:country:allowed",
    "TN",
  ],
  Array [
    "og:restrictions:country:allowed",
    "TR",
  ],
  Array [
    "og:restrictions:country:allowed",
    "TW",
  ],
  Array [
    "og:restrictions:country:allowed",
    "US",
  ],
  Array [
    "og:restrictions:country:allowed",
    "UY",
  ],
  Array [
    "og:restrictions:country:allowed",
    "VN",
  ],
  Array [
    "og:restrictions:country:allowed",
    "ZA",
  ],
  Array [
    "og:audio",
    "https://p.scdn.co/mp3-preview/d1ff0ba5c5538ca2c50b808aab2278253c98b038?cid=162b7dc01f3a4a2ca32ed3cec83d1e02",
  ],
  Array [
    "og:audio:type",
    "audio/vnd.facebook.bridge",
  ],
  Array [
    "music:preview_url:url",
    "http://p.scdn.co/mp3-preview/d1ff0ba5c5538ca2c50b808aab2278253c98b038?cid=162b7dc01f3a4a2ca32ed3cec83d1e02",
  ],
  Array [
    "music:preview_url:secure_url",
    "https://p.scdn.co/mp3-preview/d1ff0ba5c5538ca2c50b808aab2278253c98b038?cid=162b7dc01f3a4a2ca32ed3cec83d1e02",
  ],
  Array [
    "music:preview_url:type",
    "audio/mpeg",
  ],
  Array [
    "twitter:image",
    "https://i.scdn.co/image/356e29a0f112a4835f7b22f5e586a1907eae2031",
  ],
  Array [
    "twitter:audio:partner",
    "Spotify",
  ],
  Array [
    "twitter:title",
    "Bohemian Rhapsody - Remastered 2011",
  ],
  Array [
    "twitter:app:id:iphone",
    "324684580",
  ],
  Array [
    "twitter:app:id:googleplay",
    "com.spotify.music",
  ],
  Array [
    "twitter:site",
    "@spotify",
  ],
  Array [
    "twitter:description",
    "Bohemian Rhapsody - Remastered 2011, a song by Queen on Spotify",
  ],
  Array [
    "twitter:player:height",
    "584",
  ],
  Array [
    "twitter:player:width",
    "504",
  ],
  Array [
    "twitter:card",
    "audio",
  ],
  Array [
    "twitter:player",
    "https://open.spotify.com/embed/track/4u7EnebtmKWzUH433cf5Qv?utm_campaign=twitter-player&amp;utm_source=open&amp;utm_medium=twitter",
  ],
  Array [
    "twitter:audio:artist_name",
    "Queen",
  ],
  Array [
    "twitter:audio:source",
    "https://open.spotify.com/twitter/vmap/track/4u7EnebtmKWzUH433cf5Qv?utm_campaign=twitter-player&amp;utm_source=open&amp;utm_medium=twitter",
  ],
  Array [
    "al:android:app_name",
    "Spotify",
  ],
  Array [
    "al:android:package",
    "com.spotify.music",
  ],
  Array [
    "al:android:url",
    "spotify://track/4u7EnebtmKWzUH433cf5Qv",
  ],
  Array [
    "al:ios:app_name",
    "Spotify",
  ],
  Array [
    "al:ios:app_store_id",
    "324684580",
  ],
  Array [
    "al:ios:url",
    "spotify://track/4u7EnebtmKWzUH433cf5Qv",
  ],
  Array [
    "og:site_name",
    "Spotify",
  ],
  Array [
    "google",
    "notranslate",
  ],
  Array [
    "fb:app_id",
    "174829003346",
  ],
]
`;
